analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -450,6 +459,399 @@ public void Finally_NestedFinally()
+                "InFinally",
+                "InFinally");
+
+            validator.TagStates("InCatch").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            ValidateHasOnlyNoExceptionAndUnknownException(validator, "InFinally");
+        }
+
+        [TestMethod]
+        public void TryCatch_NestedThrowWithCatchFilter()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    try
+    {
+        tag = ""InNestedTry"";
+        throw new ArgumentNullException();
+        tag = ""UnreachableInNestedTry"";
+    }
+    catch
+    {
+        tag = ""InNestedCatch"";
+    }
+}
+catch (ArgumentNullException)
+{
+    tag = ""InCatch"";
+}
+tag = ""After"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InNestedTry"); // ToDo: the rest of the tags are missing since we don't support Conversion operation
+        }
+
+        [TestMethod]
+        public void TryCatch_NestedThrowWithNestedCatchFilter()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    try
+    {
+        tag = ""InNestedTry"";
+        throw new ArgumentNullException();
+        tag = ""UnreachableInNestedTry"";
+    }
+    catch (NotSupportedException)
+    {
+        tag = ""InNestedCatch"";
+    }
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""After"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InNestedTry"); // ToDo: the rest of the tags are missing since we don't support Conversion operation
+        }
+
+        [TestMethod]
+        public void TryCatch_NestedThrowWithMultipleCatchFiltersOnDifferentLevels()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    try
+    {
+        tag = ""InNestedTry"";
+        throw new ArgumentNullException();
+        tag = ""UnreachableInNestedTry"";
+    }
+    catch (NotSupportedException)
+    {
+        tag = ""InNestedCatch"";
+    }
+}
+catch (FormatException)
+{
+    tag = ""InCatch"";
+}
+tag = ""After"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InNestedTry"); // ToDo: the rest of the tags are missing since we don't support Conversion operation
+        }
+
+        private static void ValidateHasOnlyUnknownExceptionAndSystemException(ValidatorTestCheck validator, string stateName) =>
+            validator.TagStates(stateName).Should().HaveCount(2)
+                .And.ContainSingle(x => HasUnknownException(x))
+                .And.ContainSingle(x => HasExceptionOfTypeException(x));
```

Comment:
This will now make sense and remove the double-exception. Can be done in another PR.
```suggestion
                .And.ContainSingle(x => HasSystemException(x));
```